ADR/00xx-component-dependencies.md

```diff
@@ -0,0 +1,171 @@
+* [integration-service] will always skip testing for all components except for the bundle image Component, because the bundle image Component declares that it depends on all other Components. (All other Components are **dependency** Components. The bundle is the only **dependant** Component.)
+* [build-service] will propagate digest references as a PR to the bundle image, in the same PR Group
+  as the originating PR. The PR will be marked as "Draft" to indicate they should not be merged.
+* When the operand image PR is merged, [build-service] will update the PRs it originally filed to
+  take them out of "Draft" to indicate that they are safe to merge now as long as there are no other
+  unmerged triggering PRs in the same group.
+
+Think about branches:
+
+* User team updates their operand images with branch `feature-1234` (or whatever).
+* [build-service] uses that same branch in its PR to the bundle repo.
+* [build-service] uses that branch name as a prefix in the branch names that it chooses (`feature-1234/<suffix>`) to send to the bundle repo.
+* If user submits three operand image PRs and uses the same branch name for all (`feature-1234`) – then the digests all pile up on the same bundle branch (`feature-1234/<suffix>`), the same bundle PR.
+* If the user submits three operand image PRs and uses different branch names for all – then the digests are split among three different bundle branches, three different bundle PRs.
+
+---
+
+### OLM Operators, with components in the same repo
+
+Another OLM Operator Scenario: an operator git repo contains both the controller code and the bundle metadata.
+
+* The user has two Components, that both point to the same repo.
+* The bundle Component declares that it `depends on` the controller image Component, by way of the new field on the Component CR.
+* [integration-service] will always skip testing for the controller Component because it is known to be an **dependency** component.
+* [build-service] will propagate the digest reference as a PR to the bundle image, which happens to be the same git repository as the controller Component.
+
+Think about branches:
+
+* The user submitted their controller image update on the `new-feature` branch.
+* [build-service] pushes its commits with the new image digest references to a new feature branch that uses the triggering feature branch name as a prefix: `new-feature/<suffix>`.
+  * This works like it does in most other cases. Let the user merge the original PR, and only after that will [integration-service] update the checks for `new-feature` to say “okay to merge now!”
+  * When the user merges the originla PR, [build-service] will update the PR it filed on the `new-feature/<suffix>` branch to take it out of "Draft", indicating that it is safe to merge now. It may potentially rebase the PR to trigger a new build or use `/retest`.
```

Comment:
```suggestion
  * When the user merges the original PR, [build-service] will update the PR it filed on the `new-feature/<suffix>` branch to take it out of "Draft", indicating that it is safe to merge now. It may potentially rebase the PR to trigger a new build or use `/retest`.
```